mgcpy/independence_tests/abstract_class.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+
+
+class IndependenceTest(ABC):
+    """
+    IndependenceTest abstract class
+
+    Specifies the generic interface that must be implemented by
+    all the independence tests in the mgcpy package.
+
+    :param data_matrix_X: a [n*d] data matrix, a square matrix with n samples in d dimensions
+    :type data_matrix_X: 2D `numpy.array`
+
+    :param data_matrix_Y: a [n*d] data matrix, a square matrix with n samples in d dimensions
```

Comment:
X and Y don't need to have the same dimensions, i.e. X [n*p] and Y [n*q], but they do need to have the same number of examples

Reply:
otherwise looks good

Reply:
Done!